### PR TITLE
refactor,chore(egen) : refactored code according to template guidelines

### PR DESCRIPTION
**Pull request by Jayakrishna2801**
This PR includes

1.Adds Colab authentication and kernal restart steps.
2.Adds colab enterprise logo with link.
3.Removes IS_TESTING condition.
4.Replace REGION with LOCATION.
5.Removed UUID code
6.added missing code and removed unused modules
7.added required libraries in installation section

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 1]
```diff
@@ -637,7 +569,7 @@
         "id": "eAJDP_phDjOQ"
       },
       "source": [
-        "You see that there are three months of dates in this data, November 2020-January 2021, and somewhere on the order of tens of thousands of events per day (at least for the days that are visible in the default output)."
+        "You see that there are three months of dates in this data, November 2020-January 2021, and somewhere, on the order of tens of thousands of events per day (at least for the days that are visible in the default output)."
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:572]: remove comma: "and somewhere on the order of"

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 2]
```diff
@@ -847,7 +779,7 @@
       },
       "source": [
         "## Explore event counts by various categories\n",
-        "In this section, you use plotting libraries - specifically [matplotlib](https://matplotlib.org/) and [Plotly Express](https://plotly.com/python/plotly-express/) - to generate graphical outputs that help understand the distribution of events across some of the fields in the sample GA4 data."
+        "In this section, you use plotting libraries - specifically [matplotlib](https://matplotlib.org/) and [Plotly Express](https://plotly.com/python/plotly-express/) - to generate graphical outputs that helps understand the distribution of events across some of the fields in the sample GA4 data."
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:782]: add "to"
- "...to generate graphical outputs that helps to understand the distribution..."

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 3]
```diff
@@ -1150,7 +1082,7 @@
...
-        "Building on the analysis from the previous steps, you can also query and plot event distributions by the combination of two categorical variables together. You previously looked at the individual event distributions for country and operating system (OS) separately. In this case, you focus on *the combination* of country and OS together. In addition to the raw number of events in each country-OS combination, you also calculate the number as a percentage of *all events in that country*, to see if certain countries seem to have more/less activity on a given OS."
+        "Building on the analysis from the previous steps, you can also query and plot event distributions by the combination of two categorical variables together. You previously looked at the individual event distributions for country and operating system (OS) separately. In this case, you focus on the combination of country and OS together. In addition to the raw number of events in each country OS combination, you also calculate the number as a percentage of *all events in that country*, to see if certain countries seem to have more/less activity on a given OS."
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:1085]: change to "more or less"

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 4]
```diff
@@ -1239,7 +1171,7 @@
         "id": "B5HcMZfLsTJA"
       },
       "source": [
-        "From looking at the colors and hovering over some of the numbers, it's clear that the distribution of events across device OS doesn't look very different across countries. Web is the dominant OS in every country, accounting for 55-65% of events in each country analyzed here. There are some other systems with slightly higher proportions in some countries (e.g. Windows in France, iOS in Singapore), but you likely come away from the heatmap thinking how relatively consistent things look across countries."
+        "From looking at the colors and hovering over some of the numbers, it's clear that the distribution of events across device OS doesn't look very different across countries. Web is the dominant OS in every country, accounting for 55 to 65% of events in each country analyzed here. There are some other systems with slightly higher proportions in some countries (e.g. Windows in France, iOS in Singapore), but you likely come away from the heatmap thinking how relatively consistent things look across countries."
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:1174]: add comma: e.g.,

**Review by kittyabs — COMMENTED** [review]
some edits

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 5]
```diff
@@ -945,7 +870,7 @@
       },
       "source": [
         "### Function to generalize event counts bar plot creation\n",
-        "Suppose that you like the previous interactive bar plot of event counts, and want to look at it for other categorical variables in the data like those related to device, user characteristics, or other fields. A nice to way generalize this is to create a function that gets event counts data from BigQuery for any single variable and then creates an interactive bar plot like the one above. Run the following cell to create such a function."
+        "Suppose that you like the previous interactive bar plot of event counts, and want to look at it for other categorical variables in the data like those related to device, user characteristics, or other fields. A nice way to generalize this is to create a function that gets event counts data from BigQuery for any single variable and then creates an interactive bar plot like the one above. Run the following cell to create such function."
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:873]: What does "it" refer to? another interactive bar plot? Please clarify what "it" refers to.  ("...and want to look at it for other categorical variables..")

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 6]
```diff
@@ -1060,7 +985,7 @@
         "\n",
         "*   The United States is the country with the most events, followed by India and the United Kingdom.\n",
         "*   \"Web\" is the most common device operating system by far, with iOS and Windows pretty close for second. \n",
-        "*   The most common value in the device language field is actually NULL, or no data in the field. This is important to note in any sort of further analysis or modeling using language. You may want to understand why it's missing, what types of users/events have missing language more or less often, and how that may/may not change in future data."
+        "*   The most common value in the device language field is actually NULL, or no data in the field. This is important to note in any sort of further analysis or modeling using language. You may want to understand why it's missing, what types of users/events have missing language more or less often, and how that might change in future data."
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:988]: Again, clarify what "it" is referencing ("it's missing...")

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 7]
```diff
@@ -1150,7 +1075,7 @@
...
-        "Building on the analysis from the previous steps, you can also query and plot event distributions by the combination of two categorical variables together. You previously looked at the individual event distributions for country and operating system (OS) separately. In this case, you focus on *the combination* of country and OS together. In addition to the raw number of events in each country-OS combination, you also calculate the number as a percentage of *all events in that country*, to see if certain countries seem to have more/less activity on a given OS."
+        "Building on the analysis from the previous steps, you can also query and plot event distributions by the combination of two categorical variables together. You previously looked at the individual event distributions for country and operating system (OS) separately. In this case, you focus on the combination of country and OS together. In addition to the raw number of events in each country OS combination, you also calculate the number as a percentage of *all events in that country*, to see if certain countries seem to have more or less activity on a given OS."
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:1078]: - change: " You previously looked at ..."
- to: "Previously, you looked at..."

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb [hunk 8]
```diff
@@ -1748,7 +1684,7 @@
       },
       "source": [
         "## Save some outputs to Cloud Storage\n",
-        "Sometimes you may want some of your code output - tables, graphs, etc. - to persist outside of your notebook. This could be to preserve some results for when you're done with the notebook or to share with various collaborators via other mediums.\n",
+        "Sometimes, you may want some of your code output - tables, graphs, etc. - to persist outside of your notebook. This could be to preserve some results for when you're done with the notebook or to share with various collaborators via other mediums.\n",
```
Comment (kittyabs) [inline notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:1687]: - change: " output - tables, graphs, etc. - to"
- to: " output - for example,  tables and graphs - to"

**Review by kittyabs — COMMENTED** [review]
several edits and requests to clarify the referent of "it".

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb (hunk 5 above)
Comment (Jayakrishna2801) [reply notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:873]: here  , it refers to previous interactive bar plot

File: notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb (hunk 6 above)
Comment (Jayakrishna2801) [reply notebooks/official/workbench/exploratory_data_analysis/explore_data_in_bigquery_with_workbench.ipynb:988]: it referring to -->  language field , as it is null or no data in the field

**Review by katiemn — APPROVED** [review]

**Pull request merged**